Implement key-value storage with put, get and size methods.

RVoinahii opened the pull request.

RVoinahii committed:
Implement key-value storage with put, get and size methods.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/impl/StorageImpl.java:
```diff
@@ -3,17 +3,44 @@
+    private V[] valuesArray;
+    private int valuesSize;
+
+    public StorageImpl() {
```
Don't begin class or method implementation with empty line. Please remove the redundant empty line at the start of the constructor.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/impl/StorageImpl.java:
```diff
@@ -3,17 +3,44 @@
+    private int valuesSize;
+
+    public StorageImpl() {
+        keysArray = (K[]) new Object[10];
```
Any magic numbers should be constants. The size of the arrays should be defined as a constant field, not hardcoded as '10'.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/impl/StorageImpl.java:
```diff
@@ -3,17 +3,44 @@
+            }
+        }
+
+        if (valuesSize < 10) {
```
Any magic numbers should be constants. The comparison 'valuesSize < 10' uses a magic number; consider defining a constant for the maximum storage size.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/impl/StorageImpl.java:
```diff
@@ -3,17 +3,44 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private K[] keysArray;
+    private V[] valuesArray;
```
All non-constant fields should be initialized in the constructor. The fields 'keysArray' and 'valuesArray' are correctly initialized in the constructor.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/impl/StorageImpl.java:
```diff
@@ -3,17 +3,44 @@
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < valuesSize; i++) {
+            if (keysArray[i] == null ? key == null : keysArray[i].equals(key)) {
```
Create informative variable and method names. The name 'valuesSize' is a bit misleading because it represents the number of elements in both, the keys and values arrays, not just the values. Consider renaming it to 'size' or 'elementsCount' for better clarity.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/impl/StorageImpl.java:
```diff
@@ -3,17 +3,44 @@
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < valuesSize; i++) {
```
Don't use class Objects. It's better to avoid using raw array types for keys and values. Consider alternatives that prevent unchecked assignments and casts.

mateacademy-ai-mentor reviewed (changes requested):
Hey there! It looks like there are a few key areas we need to address before moving forward. Specifically, let's work on replacing the use of raw array types with safer alternatives to avoid unchecked operations. Also, renaming 'valuesSize' to something more descriptive like 'size' or 'elementsCount' will make the code clearer. Additionally, defining constants for magic numbers like the number 10 in the array sizes and comparisons will greatly improve the maintainability of the code. Keep up the good work, and with these changes, your code will be even better! 🌟

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

RVoinahii committed:
Refactor StorageImpl:
- Remove redundant empty line in constructor.
- Replace magic number with a constant for array size.
- Rename 'valueSize' to 'elementsCount' for clarity.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/impl/StorageImpl.java:
```diff
@@ -3,17 +3,45 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
```
The class implementation should not begin with an empty line. Please remove the empty line after the class declaration to adhere to the coding conventions.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/impl/StorageImpl.java:
```diff
@@ -3,17 +3,45 @@
+    private V[] valuesArray;
+    private int elementsCount;
+
+    public StorageImpl() {
```
There is an unnecessary empty line at the start of the constructor. According to the checklist, you should remove all redundant empty lines.

mateacademy-ai-mentor reviewed (approved):
Great job on your Pull Request! Just a small note for the future: try to keep your code clean by removing unnecessary empty lines, like the one at the beginning of your constructor. It's a minor issue, so I'm approving your work. Keep it up! 👍💻

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨
